### PR TITLE
Revive #62457

### PR DESCRIPTION
**Pull request by MNG-cataclysm**
#### Summary
Infrastructure "Convert the Kill Zombie Mom mission to JSON."

#### Purpose of change
For some reason, #62457 was approved, linted, cleaned, and ready, but was never merged and eventually closed due to stalebot. I thought that it was good, so I decided to resurrect it.

#### Describe the solution
See #62457 for details.

#### Describe alternatives you've considered
Not doing this, or doing what #62457 suggested.

#### Testing
Tested the mission, killed a dude's mom, and got paid for it.

#### Additional context
Instead of "Zombie Mom" as the placeholder name, I went with "Catherine Lloyd" instead. I tried using a name snippet, but apparently monster names don't support the use of name snippets. Would be pretty nice if they did, though.

### CODE REVIEW

**Comment by Maleclypse** [comment]
@lispcoc, @RenechCDDA and @ehughsbaird  any of you have a good idea on how to make name in mission easily accept either snippets or pull from the randomized name base in game? If the later respecting gendered names?

**Comment by MNG-cataclysm** [comment]
> @lispcoc, @RenechCDDA and @ehughsbaird  any of you have a good idea on how to make name in mission easily accept either snippets or pull from the randomized name base in game? If the later respecting gendered names?

With snippets, we could just take a big list of names and call it something like <female_first_name>, I suppose.

**Comment by ehughsbaird** [comment]
It would be a small C++ change here (the constructor and apply function) https://github.com/CleverRaven/Cataclysm-DDA/blob/f1521a8287467f2b81707ae01cdd01857c6234b7/src/mapgen.cpp#L2329. I don't know how names are accessed, but it would be easy to add snippet support.

**Comment by Maleclypse** [comment]
> It would be a small C++ change here (the constructor and apply function)
> 
> https://github.com/CleverRaven/Cataclysm-DDA/blob/f1521a8287467f2b81707ae01cdd01857c6234b7/src/mapgen.cpp#L2329
> 
> . I don't know how names are accessed, but it would be easy to add snippet support.

I'll make an issue for it. Thank you for letting me know if it's a big ask or not.

**Comment by ehughsbaird** [comment]
No need https://github.com/CleverRaven/Cataclysm-DDA/pull/69270

**Comment by Maleclypse** [comment]
https://github.com/CleverRaven/Cataclysm-DDA/pull/69270 has merged if you want to create a list of potential names.

**Comment by MNG-cataclysm** [comment]
> https://github.com/CleverRaven/Cataclysm-DDA/pull/69270 has merged if you want to create a list of potential names.

Alright, I'll get to it.

File: data/json/npcs/missiondef.json [hunk 1]
```diff
@@ -583,8 +583,18 @@
...
+      "assign_mission_target": {
+        "om_terrain": "house",
+        "om_terrain_match_type": "PREFIX",
+        "reveal_radius": 1,
+        "random": true,
+        "search_range": 60,
+        "min_distance": 5,
+        "z": 0
+      },
+      "update_mapgen": { "place_monster": [ { "monster": "mon_zombie", "x": 12, "y": 12, "target": true, "name": "Catherine Lloyd" } ] }
```
Comment (ehughsbaird) [inline data/json/npcs/missiondef.json:596]: This results in the same behavior as the C++ version.
```suggestion
      "update_mapgen": { "place_monster": [ { "monster": "mon_zombie", "x": 12, "y": 12, "target": true, "random_name": "female" } ] }
```

**Pull request merged**